src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnection.cs

```diff
@@ -1006,7 +1006,7 @@ private bool TryOpen(TaskCompletionSource<DbConnectionInternal> retry)
             }
 
             if (StatisticsEnabled ||
-                s_diagnosticListener.IsEnabled(SqlClientDiagnosticListenerExtensions.SqlAfterExecuteCommand))
+                (s_diagnosticListener.IsEnabled(SqlClientDiagnosticListenerExtensions.SqlAfterExecuteCommand) && Statistics != null ))
```

Comment:
_statistics can be set to Null or non-null value by setting the `SqlConnection.StatisticsEnabled` property. https://github.com/dotnet/corefx/blob/bffef76f6af208e2042a2f27bc081ee908bb390b/src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnection.cs#L82 

I will make the change. I will make a copy of `_statistics ` store it in a local and check the local.